src/unix/posix.h

```diff
@@ -80,4 +80,14 @@ GIT_INLINE(int) p_futimes(int f, const struct p_timeval t[2])
 # define p_futimes futimes
 #endif
 
+#ifdef HAVE_REGCOMP_L
+#include <xlocale.h>
+GIT_INLINE(int) p_regcomp(regex_t *preg, const char *pattern, int cflags)
+{
+	return regcomp_l(preg, pattern, cflags, (locale_t) 0);
```

Comment:
From Apple's macOS source code release:

```c
int
tre_regncomp_l(regex_t *preg, const char *regex, size_t n, int cflags, locale_t loc)
{
  ...

  NORMALIZE_LOCALE(loc);

  ...
}
```

```c
#define NORMALIZE_LOCALE(x)	if ((x) == NULL) { \
	(x) = _c_locale; \
} else if ((x) == LC_GLOBAL_LOCALE) { \
	(x) = &__global_locale; \
}
```

So we should be fine passing `(locale_t) 0` (or even just `NULL`) here. Do you want me to change anything here?